### PR TITLE
Allow closing the program with close button

### LINKED ISSUE #23: allow closing the program with close button
the close button of the window is currently not operational as it just closes _that_ window but a new one is launched in the next step again. we need to handle the event and properly terminate the program.

### PR DESCRIPTION
**Pull request by Nic822**
#23

### CODE REVIEW

File: src/app.py
Comment (rursprung) [inline src/app.py]: what is `quit`? this variable is not defined anywhere and not used anywhere

File: src/app.py
Comment (rursprung) [inline src/app.py]: this seems to be the solution we need 👍

File: src/output/tts/tts.py
Comment (rursprung) [inline src/output/tts/tts.py]: this looks wrong (and you seem to have only done this so that you can pass `quit` in `app.py` to this method call - even though `quit` is neither defined nor used)

File: src/output/tts/tts_online.py
Comment (rursprung) [inline src/output/tts/tts_online.py:22]: is this used to allow closing the program while the TTS is talking (because we're freezing the main thread for this period as we're not doing TTS in a separate thread)?

File: src/output/tts/tts_online.py
Comment (rursprung) [inline src/output/tts/tts_online.py]: this shouldn't be necessary. according to [the `pygame.quit` documentation](https://www.pygame.org/docs/ref/pygame.html#pygame.quit):

> [..] When the Python interpreter shuts down, this method is called regardless, so your program should not need it, except when it wants to terminate its pygame resources and continue. [..]
> Note: Calling `pygame.quit()` will not exit your program. Consider letting your program end in the same way a normal Python program will end.

so just a `break` here should be enough

File: src/output/visualizer/visualizer.py
Comment (rursprung) [inline src/output/visualizer/visualizer.py]: there's actually a nicer/cleaner way to do this in python:
* rename `self.name` to `self._name` (we should've anyway used that name; something i can clean up in a follow-up PR in all files)
* name this method here `name` and annotate it with `@property`. this will then allow you to use it as `visualizer.name` (instead of `visualizer.get_name()`), but it's a read-only property access instead of having read-write access to the member variable (as you'd get if you'd write `visualiser.name` right now which would also work since python doesn't have a visibility (private/public) concept)

see e.g. [this guide](https://www.geeksforgeeks.org/python-property-decorator-property/) on how `@property` works (note: here we don't need/want the setter & deleter which they also show - just the classic `@property`)

**Review by rursprung — CHANGES_REQUESTED** [review]
thanks for this!

please re-word your commit message according to [the best practices](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).

File: src/app.py
Comment (Nic822) [reply src/app.py]: forgot to delete after an idea, thanks for the hint.

File: src/output/tts/tts.py
Comment (Nic822) [reply src/output/tts/tts.py]: forgot to delete after an idea, thanks for the hint.

File: src/output/tts/tts_online.py
Comment (Nic822) [reply src/output/tts/tts_online.py:22]: yes, after closing the window, the speech is stil running for the last pattern

Comment (Nic822) [reply src/output/tts/tts_online.py:22]: yes, after closing the window, the speech for the last pattern ran.

Comment (Nic822) [reply src/output/tts/tts_online.py:22]: yes, after closing the window, the speech for the last pattern was still running to the end

**Review by Nic822 — COMMENTED** [review]
changed the proposals

File: src/output/visualizer/visualizer.py
Comment (rursprung) [inline src/output/visualizer/visualizer.py]: unnecessary comment, please remove

**Review by rursprung — CHANGES_REQUESTED** [review]
the code now looks good, thanks! please remove the mentioned comment and [adapt the commit message](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html). after that we can merge this!

**Review by rursprung — APPROVED** [review]

**Pull request merged**
